pkg/plugin/system/system.go

```diff
@@ -236,6 +251,7 @@ func (l *PluginList) insert(p plugin.Plugin) {
 
 // PluginNode is a node in the PluginList linked list.
 type PluginNode struct {
-	next   *PluginNode
-	plugin plugin.Plugin
+	next       *PluginNode
+	plugin     plugin.Plugin
+	properties []byte
```

Comment:
@f0rmiga do we want to be adding more then just `next` + `plugin` to linked-lists like this?

Reply:
Conceptually, I don't see why not. In this case, both `plugin` and `properties` form the payload of a list node. For clarity, it could be expressed like:

```go
type PluginNode struct {
    next *PluginNode
    payload struct {
        plugin plugin.Plugin
        properties []byte
    }
}
```